From 5f91db7422455762d6e00d29d92c63915d2cc586 Mon Sep 17 00:00:00 2001
From: Peter Jung <admin@ptr1337.dev>
Date: Sat, 31 Jan 2026 13:14:34 +0100
Subject: [PATCH 060/100] Revert "drm/amd/display: Fix NULL pointer dereference
 in is_hdmi_allm_mode()"

This reverts commit acf44cafcfb8439ce1d035483dae59593d93ea07.
---
 drivers/gpu/drm/amd/display/modules/info_packet/info_packet.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/display/modules/info_packet/info_packet.c b/drivers/gpu/drm/amd/display/modules/info_packet/info_packet.c
index a4edbf44a..cf5a1ccb8 100644
--- a/drivers/gpu/drm/amd/display/modules/info_packet/info_packet.c
+++ b/drivers/gpu/drm/amd/display/modules/info_packet/info_packet.c
@@ -433,8 +433,7 @@ void mod_build_vsc_infopacket(const struct dc_stream_state *stream,
 static bool is_hdmi_allm_mode(const struct dc_stream_state *stream)
 {
 	/* Sink doesn't expose ALLM support in edid */
-	if (!stream->link->local_sink ||
-	    !stream->link->local_sink->edid_caps.allm)
+	if (!stream->link->local_sink->edid_caps.allm)
 		return false;
 
 	switch (amdgpu_allm_mode) {
-- 
2.52.0

